kmod/src/dir.c

```diff
@@ -855,6 +858,10 @@ static int scoutfs_link(struct dentry *old_dentry,
 	if (ret)
 		goto out;
 
+	ret = scoutfs_orphan_dirty(sb, scoutfs_ino(inode));
```

Comment:
We only want to dirty if we're deleting the orphan.. was this not returning -ENOENT for links when we didn't have the orphan item yet?